Assets/Scripts/Game/UserInterfaceWindows/DaggerfallGuildServicePopupWindow.cs

```diff
@@ -574,14 +574,18 @@ private void DonationMsgBox_OnGotUserInput(DaggerfallInputMessageBox sender, str
             {
                 if (playerEntity.GetGoldAmount() > amount)
                 {
-                    // Deduct gold, change reputation and apply blessing
+                    // Deduct gold, and apply blessing if member
                     playerEntity.DeductGoldAmount(amount);
                     int factionId = guild.GetFactionId();
                     if (guild.IsMember() && guild.GetType() == typeof(Temple))
                         ((Temple)guild).Blessing(playerEntity, amount);     // TODO what happens when blessing is applied in classic? a message?
                     else
                         factionId = (int)Temple.GetDivine(buildingFactionId);
-                    playerEntity.FactionData.ChangeReputation(factionId, 1, true);
+
+                    // Change reputation
+                    int rep = Math.Abs(playerEntity.FactionData.GetReputation(factionId));
+                    if (UnityEngine.Random.Range(1, 100) <= (2 * amount / rep + 1))
```

Comment:
You should add 1 to the maximum to get it the same as classic, because UnityEngine.Random.Range is non-inclusive of the maximum.

Reply:
I don't believe that's the case. From the unity documentation:

Returns a random float number between and min [inclusive] and max [inclusive]

Reply:
WHy on earth is the int version an exclusive max in Unity? I had only read the float version documentation. Thanks for pointing that out.

Reply:
Maybe they expect the most common use of it to be something like UnityEngine.Random.Range(array[0], array.length).
The "int" example on https://docs.unity3d.com/ScriptReference/Random.Range.html looks like it may be the same way. I'm guessing (have not actually looked into it) that if "SceneCount" was 3, you would want to give "LoadScene" either 0, 1, or 2.